args_test.go

```diff
@@ -5,6 +5,18 @@ import (
 	"testing"
 )
 
+func getCommand(args PositionalArgs, withValid bool) *Command {
```

Comment:
Hmmm not sure if I'm a huge fan of using helper methods to abstract away creating a command struct. What benefit do we get adding this complexity when using literal structs inline gets us the same thing?

Reply:
There are two main reasons: verbosity and human abstraction.

- `get_command` is used 16 times, and it contains 9 lines. That's 144 lines of duplicated code. It's 50% of the file.
- Using this abstraction makes it easier to understand that two orthogonal features are tested: different types of argument constraints and usage of ValidArgs. The orthogonality of those features is fixed in #841. In fact, this PR is a subset of that one.

As you can see in https://github.com/spf13/cobra/pull/841/files#diff-a48034865676903aed3da7be846f56efab79615c941b4b0089d0eb17584d391a, the complexity of this file would explode in #841 if we didn't have this (and other) helper function.

Reply:
See also these two dialogues with @jharshman where he did some suggestions about renaming tests and using sub-tests, which I applied:

- https://github.com/spf13/cobra/pull/841#discussion_r267967922
- https://github.com/spf13/cobra/pull/841#discussion_r267881847

Reply:
Fair enough! I say let's get it in and if we need to refactor, we can go from there